src/RangeParser.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Pgsql;
+
+use DateInterval;
+use DateTime;
+use InvalidArgumentException;
+use function preg_match;
+
+final class RangeParser
+{
+    private const RANGES = [
+        Schema::TYPE_INT_RANGE,
+        Schema::TYPE_BIGINT_RANGE,
+        Schema::TYPE_NUM_RANGE,
+        Schema::TYPE_TS_RANGE,
+        Schema::TYPE_TS_TZ_RANGE,
+        Schema::TYPE_DATE_RANGE,
+    ];
+
+    private ?string $type;
+
+    public function __construct(?string $type = null)
+    {
+        $this->type = $type;
+    }
+
+    public function withType(?string $type): self
+    {
+        $new = clone $this;
+        $new->type = $type;
+
+        return $new;
+    }
+
+    public function asInt(): self
+    {
+        return $this->withType(Schema::TYPE_INT_RANGE);
+    }
+
+    public function asBigInt(): self
+    {
+        return $this->withType(Schema::TYPE_BIGINT_RANGE);
+    }
+
+    public function asNumeric(): self
+    {
+        return $this->withType(Schema::TYPE_NUM_RANGE);
+    }
+
+    public function asDate(): self
+    {
+        return $this->withType(Schema::TYPE_DATE_RANGE);
+    }
+
+    public function asTimestamp(): self
+    {
+        return $this->withType(Schema::TYPE_TS_RANGE);
+    }
+
+    public function asTimestampTz(): self
+    {
+        return $this->withType(Schema::TYPE_TS_TZ_RANGE);
+    }
+
+    public function asCustom(): self
+    {
+        return $this->withType(null);
+    }
+
+    public function parse(?string $value): ?array
```

Comment:
```suggestion
    public function parse(?string $value, ?string $dbType = null): ?array
```

Suggestion to pass `$dbType` to `parse()` method instead of initiating